### DISCUSSION TITLE
Do internal repos count as private, in an organization?

### DISCUSSION THREAD

**Discussion by rfalanga** [community/community#62111] [Packages] [answered]
### Select Topic Area

Question

### Body

I'm learning about GitHub Packages. I started with the Introduction to GitHub Packages. There's [one paragraph](https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages#overview-of-package-visibility) I'm not sure of, which says:

> You can publish packages in a public repository (public packages) to share with all of GitHub, or in a private repository (private packages) to share with collaborators or an organization.

At work I'm migrating our source code out of an old TFS instance into an organization in our enterprise account. In that situation we had three options when creating a new repo in the GitHub org:
- Private
- Internal
- Public

We don't want to make any of our source code public. I thought that Internal was the best choice since, as I understand it, Internal would allow all users of our company to see all the repos, without giving them access to all of them unless they're collaborators of the repo. This mimics the experience they have with the old TFS. Anyway, my question concerns the paragraph I quoted above from the documentation. Do I have to go into each repo that we'll want to have packages in to share with the rest of the organization, and change that repo's visibility from Private instead of Internal?

**Comment by CodeMaster7000** [comment, marked as answer]
Hi @rfalanga 

My knowledge says not quite. Private repos are typically only visible to you (in orgs a few other members), but internal repos are the default setting for all new repositories created in an org owned by an enterprise account. Members of the enterprise can fork any internal repository owned by an organization in the enterprise. Essentially, they are accessible to _all_ enterprise members.

However, I understand that it is only possible to create packages in private or public repos, so the safest option would be to change its visibility to private and share with all of the enterprise members.

Good luck and have a great day!

**Reply by Rod-at-DOH** [reply]
WOW, that isn't what I expected. Thank you for clarifying it!

**Reply by Rod-at-DOH** [reply]
I will, but from my other account (rfalanga) later. 😄

**Reply by rfalanga** [reply]
Thank you, @CodeMaster7000

**Reply by CodeMaster7000** [reply]
Welcome!
